src/hooks/useButtonStyles/useButtonStyles.ts

```diff
@@ -0,0 +1,29 @@
+import type { BackgroundProps, ButtonProps } from '@chakra-ui/react'
+import { useColorMode, useTheme } from '@chakra-ui/react'
+import type { SystemStyleFunction } from '@chakra-ui/theme-tools'
+import { theme } from 'theme/theme'
+
+type UseButtonStyleProps = {
+  variant: string
+  colorScheme: string
+}
+
+export const useButtonStyles = ({ variant, colorScheme }: UseButtonStyleProps) => {
+  const {
+    components: {
+      Button: { variants },
+    },
+  } = useTheme()
+
+  const { [variant]: buttonVariant } = variants as Record<string, SystemStyleFunction>
```

Comment:
`useTheme()` isn't really typed, there's a generic for it but all it does is taking a full dictionary object to be used as a return type
https://github.com/chakra-ui/chakra-ui/blob/3e1b3f6b6a7398b71ac08339110f075695fbae94/packages/components/system/src/use-theme.ts#L11

Reply:
I was just about to ask re: this, ty